gflownet.py

```diff
@@ -673,7 +677,7 @@ def sample_batch(
                 )
             t0_a_envs = time.time()
             # Add to batch
-            for env, action, valid in zip(envs, actions, valids):
+            for env, action, mask, valid in zip(envs, actions, masks, valids):
```

Comment:
I don't think we want to return the `masks` from `forward_sample()`. Reason: 
We enter `forward_sample()` at `t=0` with `env.state = s0` and get invalid mask `m0`. Then we perform the `step()` function (inside `forward_sample` so `env.state` gets updated to `s1`. But we don't find the new mask `m1`, so what gets returned is `env` corresponding to `s1` but `mask` corresponding to `s0`, i.e., `m0`. I think we should compute the masks in the `sample_batch()` after getting the updated `env` from `forward_sample()`